Dockerfile

```diff
@@ -112,6 +118,10 @@ ARG IMGPKG_PASSWORD
 ENV IMGPKG_USERNAME=${IMGPKG_USERNAME} IMGPKG_PASSWORD=${IMGPKG_PASSWORD}
 RUN --mount=type=bind,readwrite \
     --mount=from=carvel-base,src=/bin/imgpkg,target=/bin/imgpkg \
+    --mount=from=cli-plugin-builder-install,src=/usr/bin/tanzu,target=/bin/tanzu \
+    --mount=from=cli-plugin-builder-install,src=/root/.local/share/tanzu-cli/builder,target=/root/.local/share/tanzu-cli/builder \
+    --mount=from=cli-plugin-builder-install,src=/root/.config/tanzu/,target=/root/.config/tanzu/ \
+    --mount=from=cli-plugin-builder-install,src=/root/.cache/tanzu/,target=/root/.cache/tanzu/ \
     tanzu builder plugin publish-package \
```

Comment:
As part of this command execution, the artifacts are pushed to `./build/artifacts/packages` and that was the reason for removing build directory (in the previous PR) from `.dockerignore` since we need the artifacts in the context. However, this has an implication on cache invalidation during image build process. This is the scenario:

1. As part of docker-build-all target, we run lint, test, build targets.
2. The test target generates code coverage reports and pushed it to `build/{component}/cover.out`
3. Since contents of the build directory have changed, this invalidates docker build cache for subsequent stages (i.e., build)

We can avoid this by selectively allowing only `./build/artifacts` in the docker build context. So adding these lines to the `.dockerignore` should fix this (Ref: https://docs.docker.com/engine/reference/builder/#dockerignore-file):
```
build/
!build/artifacts
```

Reply:
That suggestion sounds good to me. I'll get the change in, thanks!